### DISCUSSION TITLE
Consulta Rekomendashi

### DISCUSSION THREAD

**Discussion by ezecastiglione18** [dds-jv/foro#335] [Consultas de UI / REST] [answered]
Buenas! Tengo unas consultas en la parte de interfaz REST. Me piden que la info se debe realizar por pasos.

1) Para la parte de las preferencias, como hago para que en el caso de que se seleccionen 2 o mas opciones mandarlo eso por body del formulario? 

2) Primero tenia en mente que el usuario seleccione las preferencias, despues pasa a otra ventana donde seleccione el tipo de pedido y despues (segun el tipo de pedido) te redirige a una pagina o a otra. En el caso de que paso de las ventanas de preferencias a la de tipo de pedido, tengo que mandar tambien lo que selecciono?

3) Existe alguna chance de que se pueda meter alguna explicacion o posible solucion para ese parcial? Porque la que esta en la catedra (sin faltarle el respeto al que resolvio el parcial) esta desaprobado

Saludos y gracias!

**Comment by gastonprieto** [comment, marked as answer]
Hola @ezecastiglione18 

1. Si vos elegis un input de tipo [checkbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox) entonces es de elección múltiple, y eso viaja como un array de valores (en formato url encoded) 
2. Cosa importante, donde dice ventana, debería ser página, en los ejercicios no abrimos ventanas (popus o similares) sino que vamos entrando a distintas paginas con links o envios de formularios, por lo tanto las pantallas que dibujes tienen que respetar esto. No recuerdo totalmente el parcial, pero si tenias que hacer el wizard entonces tenes varias partes y mantener un estados hasta llegar al paso final. Se te ocurre como?

<details>
  <summary>Spoiler warning</summary>
  ¿Cómo sabes que el usuario esta logueado en todas las pantallas?
</details>

**Reply by ezecastiglione18** [reply]
Osea, vos tenes las sessions para saber si el usuario esta logueado. Lo que se me ocurre es que como que la info se va pasando de pagina en pagina... O a lo sumo ir guardandolo en el controller. Pero de que te sirve saber si el usuario esta logueado en todas las pantallas? Porque el mismo usuario puede hacer multiples pedidos y no se como apliicar las sessions ahi.

Pd: aca te dejo el [link](https://docs.google.com/document/d/1OTmrCpW-Ode-h_k1qWIE9OB4Uuzs9ZbR_A1JMJwHK-I/edit#heading=h.l1o68lkpuj7d) al parcial
